python/cudf/cudf/testing/testing.py

```diff
@@ -0,0 +1,600 @@
+        Compare datetime-like which is comparable ignoring dtype.
+    check_categorical : bool, default True
+        Whether to compare internal Categorical exactly.
+    check_like : bool, default False
+        If True, ignore the order of index & columns.
+        Note: index labels must match their respective
+        rows (same as in columns) - same labels must be with the same data.
+    obj : str, default ‘DataFrame’
+        Specify object name being compared, internally used to
+        show appropriate assertion message.
+
+    Examples
+    --------
+    >>> import cudf
+    >>> df1 = cudf.DataFrame({"a":[1, 2], "b":[1.0, 2.0]}, index=[1, 2])
+    >>> df2 = cudf.DataFrame({"a":[1, 2], "b":[1.0, 2.0]}, index=[2, 3])
+    >>> cudf.testing.assert_frame_equal(df1, df2)
+    ......
+    ......
+    AssertionError: ColumnBase are different
+
+    values are different (100.0 %)
+    [left]:  [1 2]
+    [right]: [2 3]
+
+    >>> df2 = cudf.DataFrame({"a":[1, 2], "c":[1.0, 2.0]}, index=[1, 2])
+    >>> cudf.testing.assert_frame_equal(df1, df2)
+    ......
+    ......
+    AssertionError: DataFrame.columns are different
+
+    DataFrame.columns values are different (50.0 %)
+    [left]: Index(['a', 'b'], dtype='object')
+    right]: Index(['a', 'c'], dtype='object')
+
+    >>> df2 = cudf.DataFrame({"a":[1, 2], "b":[1.0, 3.0]}, index=[1, 2])
+    >>> cudf.testing.assert_frame_equal(df1, df2)
+    ......
+    ......
+    AssertionError: Column name="b" are different
+
+    values are different (50.0 %)
+    [left]:  [1. 2.]
+    [right]: [1. 3.]
+
+    # This will pass without any hitch
+    >>> df2 = cudf.DataFrame({"a":[1, 2], "b":[1.0, 2.0]}, index=[1, 2])
+    >>> cudf.testing.assert_frame_equal(df1, df2)
+    """
+    _check_isinstance(left, right, cudf.DataFrame)
+
+    if check_frame_type:
+        assert isinstance(left, type(right))
+
+    # shape comparison
+    if left.shape != right.shape:
+        raise AssertionError("left and right shape mismatch")
+
+    if check_like:
+        left, right = left.reindex(index=right.index), right
+        right = right[list(left._data.names)]
+
+    if check_less_precise:
+        raise NotImplementedError("check_less_precise is not yet supported")
+
+    # index comparison
+    assert_index_equal(
+        left.index,
+        right.index,
+        exact=check_index_type,
+        check_names=check_names,
+        check_less_precise=check_less_precise,
+        check_exact=check_exact,
+        check_categorical=check_categorical,
+        obj=f"{obj}.index",
+    )
+
+    pd.testing.assert_index_equal(
+        left.columns,
+        right.columns,
+        exact=check_index_type,
+        check_names=check_names,
+        check_less_precise=check_less_precise,
+        check_exact=check_exact,
+        check_categorical=check_categorical,
+        obj=f"{obj}.columns",
+    )
+
+    if by_blocks:
+        raise NotImplementedError("by_blocks is not supported")
```

Comment:
A `NotImplementedError` indicates that it may be implemented in the future, let's just ignore whatever value is provided.